src/task_impl/atomic_task.rs

```diff
@@ -52,30 +70,35 @@ impl AtomicTask {
         }
     }
 
-    /// The caller must ensure mutual exclusion
-    pub unsafe fn park(&self) {
-        if let Some(ref task) = *self.task.get() {
```

Comment:
Wasn't this one of the big use cases of this cell? Is the perf impact not measureable?